packages/text-annotator/src/utils/debounce.ts

```diff
@@ -1,8 +1,22 @@
-export const debounce = <T extends (...args: any[]) => void>(func: T, delay = 10): T => {
-	let timeoutId: ReturnType<typeof setTimeout>;
+import libDebounce from 'debounce';
 
-	return ((...args: any[]) => {
-		clearTimeout(timeoutId);
-		timeoutId = setTimeout(() => func.apply(this, args), delay);
-	}) as T;
+/**
+ * Wraps the `debounce` function from the `debounce` package
+ * to make it context-agnostic.
+ * Otherwise, we won't be able to use it in multiple event listeners simultaneously,
+ * like `window.onresize` and `ResizeObserver`.
+ * @see https://github.com/sindresorhus/debounce/issues/8#issuecomment-2321341074
+ */
+export const debounce: typeof libDebounce = (function_, wait = 10, options) => {
```

Comment:
I've just addressed the issue in the `debounce` library - https://github.com/sindresorhus/debounce/pull/43. So I'm waiting for the approval and merge 🙏🏻